common/lib/xmodule/xmodule/js/src/problem/edit.coffee

```diff
@@ -552,20 +537,22 @@ class @MarkdownEditingDescriptor extends XModule.Descriptor
       });
 
       // split scripts and preformatted sections, and wrap paragraphs
-      splits = xml.split(/(\<\/?(?:script|pre).*?\>)/g);
-      scriptFlag = false;
+      splits = xml.split(/(\<\/?(?:script|pre|label|description).*?\>)/g);
+
+      // used to check if a text string be wrapped by <p> tag  
+      wrappable = true;
 
       for (i = 0; i < splits.length; i += 1) {
-          if(/\<(script|pre)/.test(splits[i])) {
-              scriptFlag = true;
+          if(/\<(script|pre|label|description)/.test(splits[i])) {
+              wrappable = false;
           }
 
-          if(!scriptFlag) {
+          if(wrappable) {
               splits[i] = splits[i].replace(/(^(?!\s*\<|$).*$)/gm, '<p>$1</p>');
           }
 
-          if(/\<\/(script|pre)/.test(splits[i])) {
-              scriptFlag = false;
+          if(/\<\/(script|pre|label|description)/.test(splits[i])) {
+              wrappable = true;
```

Comment:
How does this condition differ from lines 546-547 (where wrappable is the opposite)?